app/src/main/java/com/hedvig/app/feature/offer/usecase/ExternalInsuranceDataCollectionUseCase.kt

```diff
@@ -0,0 +1,27 @@
+package com.hedvig.app.feature.offer.usecase
+
+import com.apollographql.apollo.ApolloClient
+import com.hedvig.android.owldroid.graphql.DataCollectionResultQuery
+import com.hedvig.app.util.apollo.QueryResult
+import com.hedvig.app.util.apollo.safeQuery
+
+class ExternalInsuranceDataCollectionUseCase(
+    private val apolloClient: ApolloClient,
+) {
+    sealed class Result {
+        abstract val referenceUuid: String
+
+        data class Success(override val referenceUuid: String, val data: DataCollectionResultQuery.Data) : Result()
```

Comment:
Map `DataCollectionResultQuery.Data` to another data class instead so we dont depend on GraphQL in the viewmodel